### PR TITLE
adjust CKV_AWS_144 to support v4 changes

### LINKED ISSUE #2845: CKV_AWS_144 false negative after updating to AWS Provider 4.0
**Describe the issue**
After updating our AWS Provider to ~>4.0, we started getting a failure on `CKV_AWS_144` in our bucket module, despite having a properly configured `aws_s3_bucket_lifecycle_configuration` block.

**Examples**
Sample code:

```hcl
provider "aws" {
  alias  = "aws-primary"
  region = "us-east-1"
}

provider "aws" {
  alias  = "aws-dr"
  region = "us-west-2"
}

resource "aws_s3_bucket" "test_bucket" {
  bucket = var.bucket_name
}

resource "aws_s3_bucket" "test_dr_bucket" {
  provider = aws.aws-dr
  bucket   = "${var.bucket_name}-dr"
}

resource "aws_s3_bucket_versioning" "test_bucket_versioning" {
  bucket = aws_s3_bucket.test_bucket.id
  versioning_configuration {
    status = "Enabled"
  }
}

resource "aws_s3_bucket_versioning" "test_dr_bucket_versioning" {
  provider = aws.aws-dr
  bucket = aws_s3_bucket.test_dr_bucket.id
  versioning_configuration {
    status = "Enabled"
  }
}

resource "aws_iam_role" "dr_replication" {
  name_prefix = "replication"
  description = "Allow S3 to assume the role for replication"

  assume_role_policy = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "s3ReplicationAssume",
      "Effect": "Allow",
      "Principal": {
        "Service": "s3.amazonaws.com"
      },
      "Action": "sts:AssumeRole"
    }
  ]
}
POLICY
}

resource "aws_iam_policy" "dr_replication" {
  name_prefix = "replication"
  description = "Allows reading for replication."

  policy = <<POLICY
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Action": [
        "s3:GetReplicationConfiguration",
        "s3:ListBucket"
      ],
      "Effect": "Allow",
      "Resource": [
        "${aws_s3_bucket.test_bucket.arn}"
      ]
    },
    {
      "Action": [
        "s3:GetObjectVersion",
        "s3:GetObjectVersionForReplication",
        "s3:GetObjectVersionAcl"
      ],
      "Effect": "Allow",
      "Resource": [
        "${aws_s3_bucket.test_bucket.arn}/*"
      ]
    },
    {
      "Action": [
        "s3:ReplicateObject",
        "s3:ReplicateTags",
        "s3:ObjectOwnerOverrideToBucketOwner"
      ],
      "Effect": "Allow",
      "Resource": "${aws_s3_bucket.test_dr_bucket.arn}/*"
    }
  ]
}
POLICY
}

resource "aws_iam_policy_attachment" "dr_replication" {
  name       = "replication"
  roles      = [aws_iam_role.dr_replication.name]
  policy_arn = aws_iam_policy.dr_replication.arn
}

resource "aws_s3_bucket_replication_configuration" "dr_bucket_replication" {

  # Must have bucket versioning enabled first
  depends_on = [
    aws_s3_bucket_versioning.test_bucket_versioning,
    aws_s3_bucket_versioning.test_dr_bucket_versioning,
  ]

  role   = aws_iam_role.dr_replication.arn
  bucket = aws_s3_bucket.test_bucket.id

  rule {
    id     = "entire_bucket"
    status = "Enabled"

    destination {
      bucket        = aws_s3_bucket.test_dr_bucket.arn
      storage_class = "DEEP_ARCHIVE"
    }
  }
}

```

Expected: `CKV_AWS_144` will pass.

Actual: 
```
Check: CKV_AWS_144: "Ensure that S3 bucket has cross-region replication enabled"
        FAILED for resource: aws_s3_bucket.test_bucket
        File: /dr_test.tf:11-13
        Guide: https://docs.bridgecrew.io/docs/ensure-that-s3-bucket-has-cross-region-replication-enabled

                11 | resource "aws_s3_bucket" "test_bucket" {
                12 |   bucket = var.bucket_name
                13 | }
```

**Desktop (please complete the following information):**
 - OS: MacOS 10.14.6
 - Checkov Version 2.0.1074

**Additional context**
On the surface, this looks like related to  https://github.com/bridgecrewio/checkov/issues/2399 and https://github.com/bridgecrewio/checkov/pull/2724, but to the `CKV_AWS_144` rule.

### PR DESCRIPTION
**Pull request by gruebel**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Fixes #2845

### CODE REVIEW

File: checkov/terraform/checks/graph_checks/aws/S3BucketReplicationConfiguration.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
+metadata:
+  name: "Ensure that S3 bucket has cross-region replication enabled"
+  category: "GENERAL_SECURITY"
+  id: "CKV_AWS_144"
+definition:
+  or:
+    - cond_type: attribute
+      resource_types:
+        - aws_s3_bucket
+      attribute: replication_configuration.rules.*.status
```
Comment (achiar99) [inline checkov/terraform/checks/graph_checks/aws/S3BucketReplicationConfiguration.yaml:10]: @gruebel why do u need the `*`?

**Review by achiar99 — COMMENTED** [review]
@gruebel very nice! but i have one question
In the original check we inspect the `role` attribute, and now you inspect the `rule.status`
This is the correct way to check this violations?

**Comment by gruebel** [comment]
@achiar99 I don't why we previously checked the existence of the role, because the rule status is the important one. If the status is `Disabled` then it doesn't matter, if you attached a role, because it will no replicate anything 😄

File: checkov/terraform/checks/graph_checks/aws/S3BucketReplicationConfiguration.yaml (hunk 1 above)
Comment (gruebel) [reply checkov/terraform/checks/graph_checks/aws/S3BucketReplicationConfiguration.yaml:10]: you can actually define several different rules. So, I'm actually not sure, what would be better to check, if one of them is `Enabled` or all of them. On the other hand, why should I create a rule, which is `Disabled` 🤷  so I decided to add the wildcard here 🙂

**Review by achiar99 — DISMISSED** [review]

File: checkov/terraform/checks/graph_checks/aws/S3BucketReplicationConfiguration.yaml (hunk 1 above)
Comment (achiar99) [reply checkov/terraform/checks/graph_checks/aws/S3BucketReplicationConfiguration.yaml:10]: @gruebel  first i see in tf v4 doc that the key is `rule`. and its an object, not a list

Comment (gruebel) [reply checkov/terraform/checks/graph_checks/aws/S3BucketReplicationConfiguration.yaml:10]: correct, this one is the old way to do it, which used `rules` and in the new resource `aws_s3_bucket_replication_configuration` it is rule, which you can find at the bottom of the file. As the docs of the resource describe

> [rule](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_replication_configuration#rule) - (Required) List of configuration blocks describing the rules managing the replication [documented below](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_replication_configuration#rule).

so you can add multiple `rule` blocks, if you want 🙂

Comment (gruebel) [reply checkov/terraform/checks/graph_checks/aws/S3BucketReplicationConfiguration.yaml:10]: and here for the old way to do it directly in the `aws_s3_bucket` resource

> [rules](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket#rules) - (Required) Specifies the rules managing the replication ([documented below](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket#rules)).

pretty confusing 😄

**Review by achiar99 — APPROVED** [review]

**Pull request merged**